state/execution.go

```diff
@@ -54,30 +53,25 @@ func execBlockOnProxyApp(txEventPublisher types.TxEventPublisher, proxyAppConn p
 			// TODO: make use of this info
 			// Blocks may include invalid txs.
 			// reqDeliverTx := req.(abci.RequestDeliverTx)
-			txError := ""
 			txResult := r.DeliverTx
 			if txResult.Code == abci.CodeType_OK {
 				validTxs++
 			} else {
 				logger.Debug("Invalid tx", "code", txResult.Code, "log", txResult.Log)
 				invalidTxs++
-				txError = txResult.Code.String()
 			}
 
-			abciResponses.DeliverTx[txIndex] = txResult
-			txIndex++
-
 			// NOTE: if we count we can access the tx from the block instead of
 			// pulling it from the req
-			event := types.EventDataTx{
-				Height: block.Height,
+			txEventPublisher.PublishEventTx(types.EventDataTx{types.TxResult{
+				Height: uint64(block.Height),
+				Index:  uint32(txIndex),
 				Tx:     types.Tx(req.GetDeliverTx().Tx),
-				Data:   txResult.Data,
-				Code:   txResult.Code,
-				Log:    txResult.Log,
-				Error:  txError,
-			}
-			txEventPublisher.PublishEventTx(event)
+				Result: *txResult,
+			}})
+
+			abciResponses.DeliverTx[txIndex] = txResult
```

Comment:
I think we can remove DeliverTx field from `abciResponses`, no?

Reply:
I believe we still need them for replay if we crash after calling abciApp.Commit() but before saving the state.

https://github.com/tendermint/tendermint/blob/master/consensus/replay.go#L381

Reply:
oh, right